packages/adapter-utils/src/utils.ts

```diff
@@ -166,19 +169,26 @@ const fieldMapperGenerator = (type: ObjectType | TypeMap | MapType | ListType, v
 const removeEmptyParts = ({
   value,
   allowEmptyArrays,
-  allowEmptyObjects,
+  allowExistingEmptyObjects,
+  allowAllEmptyObjects,
 }: {
   value: Value
   allowEmptyArrays: boolean
-  allowEmptyObjects: boolean
+  allowExistingEmptyObjects: boolean
+  allowAllEmptyObjects: boolean
 }): Value => {
   if (Array.isArray(value)) {
     const filtered = value.filter(isDefined)
     return filtered.length === 0 && (value.length > 0 || !allowEmptyArrays) ? undefined : filtered
   }
   if (_.isPlainObject(value)) {
     const filtered = _.omitBy(value, _.isUndefined)
-    return _.isEmpty(filtered) && (!_.isEmpty(value) || !allowEmptyObjects) ? undefined : filtered
+    const isExistingEmptyObject = _.isEmpty(filtered) && _.isEmpty(value)
+    const isNewEmptyObject = _.isEmpty(filtered) && !_.isEmpty(value)
+    return (isExistingEmptyObject && !allowExistingEmptyObjects && !allowAllEmptyObjects) ||
+      (isNewEmptyObject && !allowAllEmptyObjects)
+      ? undefined
+      : filtered
```

Comment:
Feels like the logic can be more straight-forward, e.g
```suggestion
    const isEmptyAllowed = allowAllEmptyObjects || (allowExistingEmptyObjects && _.isEmpty(value))
    return _.isEmpty(filtered) && !isEmptyAllowed : undefined : filtered
```